protocol/contracts/v2/TempleDebtToken.sol

```diff
@@ -0,0 +1,583 @@
+     */
+    function _compoundedDebtorInterest(Debtor storage debtor) internal view returns (uint256) {
+        uint256 _rate = debtor.rate;
+        if (_rate == 0) return 0;
+
+        uint256 _timeElapsed = block.timestamp - debtor.checkpointTime;
+        uint256 _principal = debtor.principal;
+        uint256 _principalAndInterest = _principal + debtor.checkpoint;
+        return _principalAndInterest.continuouslyCompounded(_timeElapsed, uint96(_rate)) - _principal;
+    }
+
+    /**
+     * @notice Returns the amount of tokens owed by the debtor as of this block.
+     * It includes the principal + the base interest + specific debtor risk premium interest
+     */
+    function balanceOf(address _debtor) public override view returns (uint256) {
+        return _balanceOf(debtors[_debtor], _compoundedBaseInterest());
+    }
+
+    /**
+     * @notice Returns the amount of tokens owed by the debtor as of this block.
+     * It includes the principal + the base interest + specific debtor risk premium interest
+     */
+    function _balanceOf(Debtor storage debtor, uint256 _totalPrincipalAndBase) internal view returns (uint256) {
+        return (
+            _sharesToDebt(debtor.baseShares, _totalPrincipalAndBase, baseShares, false) + 
+            _compoundedDebtorInterest(debtor)
+        );
+    }
+
+    /**
+     * @notice The current debt for a given user split out by
+     * principal, base interest, risk premium (per debtor) interest
+     */
+    function currentDebtOf(address _debtor) external override view returns (
+        uint256 principal, 
+        uint256 baseInterest, 
+        uint256 riskPremiumInterest
+    ) {
+        Debtor storage debtor = debtors[_debtor];
+        principal = debtor.principal;
+        baseInterest = _sharesToDebt(debtor.baseShares, _compoundedBaseInterest(), baseShares, false) - principal;
+        riskPremiumInterest = _compoundedDebtorInterest(debtor);
+    }
+
+    /**
+      * @notice The current total principal + total base interest, total (estimate) debtor specific risk premium interest owed by all debtors.
+      * @dev Note the (total principal + total base interest) portion is up to date.
+      * However the (debtor specific risk premium interest) portion is likely stale.
+      * The `estimatedTotalRiskPremiumInterest` is only updated when each debtor checkpoints, so it's going to be out of date.
+      * For more up to date current totals, off-chain aggregation of balanceOf() will be required - eg via subgraph.
+      */
+    function currentTotalDebt() public override view returns (
+        uint256 principal,
+        uint256 baseInterest, 
+        uint256 estimatedRiskPremiumInterest
+    ) {
+        uint256 _basePrincipalAndInterest = _compoundedBaseInterest();
+        return (
+            totalPrincipal,
+            _basePrincipalAndInterest - totalPrincipal,
+            estimatedTotalRiskPremiumInterest
+        );
+    }
+
+    /**
```

Comment:
Hmm actually - i'm not sure this will be that useful thinking about it.

The function would have to take the debtor to checkpoint. It can update that debtor's custom interest, but all the others will still be out of date.

So it still wouldn't give an accurate number. *all* users would need to be updated.

Reply:
alright, offline sounds good then